File: database-driver-plugin/src/main/java/de/lino/database/database/sql/sqlite/SQLiteDatabaseProvider.java
package de.lino.database.database.sql.sqlite;

import de.lino.database.database.auth.Credentials;
import de.lino.database.database.DatabaseType;
import de.lino.database.database.sql.SQLDatabaseProvider;
import de.lino.database.database.sql.SQLExecution;
import lombok.Getter;
import org.jetbrains.annotations.NotNull;

/**
 * The {@link SQLDatabaseProvider} for SQLite, connected via {@link SQLExecution} using
 * {@link DatabaseType#SQLITE}'s JDBC driver and file-based URL scheme.
 */
@Getter
public class SQLiteDatabaseProvider extends SQLDatabaseProvider {

    /**
     * Connects to a SQLite database file with {@code credentials}.
     *
     * @param credentials the login credentials and connection details to connect with
     */
    public SQLiteDatabaseProvider(@NotNull Credentials credentials) {
        super(DatabaseType.SQLITE, new SQLExecution(DatabaseType.SQLITE, credentials));
    }
    
}

File: python/database-driver-plugin/src/database_driver/plugin/database/sql/sqlite/sqlite_database_provider.py
"""Mirror of ``de.lino.database.database.sql.sqlite.SQLiteDatabaseProvider``."""

from __future__ import annotations

from database_driver.api.database.auth.credentials import Credentials
from database_driver.api.database.database_type import DatabaseType

from database_driver.plugin.database.sql.sql_database_provider import SQLDatabaseProvider
from database_driver.plugin.database.sql.sql_execution import SQLExecution


class SQLiteDatabaseProvider(SQLDatabaseProvider):
    """The ``SQLDatabaseProvider`` for SQLite (file-based), connected via ``SQLExecution`` using
    ``DatabaseType.SQLITE``'s driver and connection scheme."""

    def __init__(self, credentials: Credentials) -> None:
        super().__init__(DatabaseType.SQLITE, SQLExecution(DatabaseType.SQLITE, credentials))
